NerdDinner/Controllers/DinnersController.cs

```diff
@@ -1,150 +1,150 @@
-
-                dinnerRepository.Save();
-
-                return RedirectToAction("Details", new { id = dinner.DinnerID });
-            }
-
-            catch 
-            {
-                foreach(var issue in dinner.GetRuleViolations())
-                {
-                    ModelState.AddModelError(issue.PropertyName, issue.ErrorMessage);
-                }
-            }
-
-            return View(dinner);
-        }
-        // creates a new form for a dinner
-        public ActionResult Create()
-        {
-            Dinner dinner = new Dinner() { EventDate = DateTime.Now.AddDays(7) };
-
-            return View(dinner);
-        }
-        
-        //Saves and adds dinner to db
-        [AcceptVerbs(HttpVerbs.Post)]
-        public ActionResult Create(Dinner dinner)
-        {
-            if (ModelState.IsValid)
-            {
-                try
-                {
-                    dinner.Host = "SomeUser";
-
-                    dinnerRepository.Add(dinner);
-                    dinnerRepository.Save();
-
-                    return RedirectToAction("Details", new { id = dinner.DinnerID });
-                }
-                catch
-                {
-                    foreach (var issue in dinner.GetRuleViolations())
-                    {
-                        ModelState.AddModelError(issue.PropertyName, issue.ErrorMessage);
-                    }
-                }
-            }
-            
-            return View(dinner);
-        }
-
-        //Delete a dinner
-        public ActionResult Delete(int id)
-        {
-            Dinner dinner = dinnerRepository.GetDinner(id);
-
-            if (dinner == null)
-            {
-                return View("NotFound");
-            }
-            else
-            {
-                return View(dinner);
-            }
-        }
-
-        //Saves and removes the dinner from db
-        [AcceptVerbs(HttpVerbs.Post)]
-        public ActionResult Delete(int id, string confirmButton)
-        {
-            Dinner dinner = dinnerRepository.GetDinner(id);
-
-            if (dinner == null)
-            {
-                return View("NotFound");
-            }
-
-            dinnerRepository.Delete(dinner);
-            dinnerRepository.Save();
-
-            return View("Deleted");
-        }
-    }
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using NerdDinner.Models;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+using PagedList;
+
+namespace NerdDinner.Controllers
+{
+    public class DinnersController : Controller
+    {
+
+        DinnerRepository dinnerRepository = new DinnerRepository();
+        private NerdDinnerDataContext db = new NerdDinnerDataContext();
+        const int pageSize = 10;
+
+        // HTTP-GET: /Dinners/
+        public ActionResult Index(int? page)
+        {
+            int pageIndex = page ?? 1;
+
+            var dinners = dinnerRepository.FindUpcomingDinners();

```

Comment:
Changed this to call the method in the repository that does the exact same thing.